File: python/problem-4.py
def is_palindromic(num):
    return str(num) == str(num)[::-1]

def palindromic_products():
    results = []
    for i in range(999,0,-1):
        for j in range(999,0,-1):
            if is_palindromic(i*j): results.append(i*j)
    return results

max(palindromic_products()) # => 906609

File: clojure/problem_4.clj
(require '[clojure.string :as s])

(defn palindromic? [number]
  (let [word (str number)]
    (= word (s/reverse word))))

(def palindromics
  (for [x (range 999 99 -1) y (range x 99 -1)
        :let [prod (* x y)]
        :when (palindromic? prod)]
   prod))

(println (apply max palindromics))
